perma_web/perma/tasks.py

```diff
@@ -266,32 +270,56 @@ def scroll_browser(browser):
         # URLError: the headless browser has gone away for some reason.
         pass
 
-def run_in_frames(link, browser, func, output_collector=None):
+def get_page_source(browser):
+    """ 
+        Get page source.
+        Use JS rather than browser.page_source so we get the parsed, properly formatted DOM instead of raw user HTML.
+    """
+    return browser.execute_script("return document.documentElement.outerHTML")
+
+def parse_page_source(source):
+    """
+        Return page source as a parsed PyQuery object for querying.
+         
+        PyQuery here works the same as `$(source)` in jQuery. So for example you can do `parsed_source(selector)` 
+        with the returned value to get a list of LXML elements matching the selector.
+    """
+    return PyQuery(source, parser='html')
+
+def get_dom_tree(browser):
+    with browser_running(browser):
+        return parse_page_source(get_page_source(browser))
```

Comment:
Your description of the pull request expresses concern this is slow: on 15 captures of a particular NYTimes, the longest runtime was 0.4s (post-onload), and the shortest was 0.02s (pre-onload). (Very few requests made it to the post-onload event before crashing). On 5 captures of particular Forbes article, the longest runtime was 0.1s (post-onload), and the shortest was 0.02s (pre-onload)) That seems plenty fast for government work ;-)